apps/api/tasks.py

```diff
@@ -0,0 +1,57 @@
+import commonware
+import json
+
+from amo.celery import task
+from apps.api.github import GithubCallback, rezip_file
+from apps.files.models import FileUpload
+from apps.devhub.tasks import validate
+
+log = commonware.log.getLogger('z.github')
+
+
+@task
+def process_webhook(upload_pk, callbacks):
+    log.info('Processing webhook for: {}'.format(upload_pk))
+    upload = FileUpload.objects.get(pk=upload_pk)
+    github = GithubCallback(callbacks)
+    res = github.get()
+
+    upload.name = '{}-github-webhook.xpi'.format(upload.pk)
+    upload.path = rezip_file(res, upload.pk)
+    upload.save()
+
+    log.info('Validating: {}'.format(upload_pk))
+    validate(
+        upload,
+        listed=True,
+        subtask=process_results.si(upload_pk, callbacks)
+    )
+
+
+@task
+def process_results(upload_pk, callbacks):
+    log.info('Processing validation results for: {}'.format(upload_pk))
+    upload = FileUpload.objects.get(pk=upload_pk)
+    validation = json.loads(upload.validation) if upload.validation else {}
+    github = GithubCallback(callbacks)
+
+    if not validation:
+        log.error('Validation not written: {}'.format(upload_pk))
+        github.failure()
+        return
+
+    if validation.get('success'):
+        log.info('Notifying success for: {}'.format(upload_pk))
+        github.success()
+        return
+
+    log.info('Notifying errors for: {}'.format(upload_pk))
+    for error in validation.get('messages', []):
```

Comment:
Yeah I was wondering about this, because by default there's a limit to 5,000 API requests an hour, I could see one add-on just blowing through our entire API queue. Know an add-on off hand that I could test this on?